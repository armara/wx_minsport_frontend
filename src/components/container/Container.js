import React from 'react'
import PropTypes from 'prop-types'
import classnames from 'classnames'

import ToolTipIcon from 'components/toolTip/icon/ToolTipIcon'

import './Container.scss'

const Container = ({ title, className, children, dataTip }) => (
  <div className={classnames(['base-container', className])}>
    {title && (
      <h5 className="base-container_title">
        {title}
        {dataTip && <ToolTipIcon dataTip={dataTip} />}
      </h5>
    )}
    {children}
  </div>
)

Container.defaultProps = {
  title: '',
  className: '',
  dataTip: '',
}

Container.propTypes = {
  title: PropTypes.string,
  className: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
    .isRequired,
  dataTip: PropTypes.string,
}

export default Container
