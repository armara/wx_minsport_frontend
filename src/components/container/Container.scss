@import 'styles/mixins';

.base-container {
  display: flex;
  flex-flow: column;
  padding-left: 20px;
  padding-right: 20px;
  padding-bottom: 20px;
  border: 1px solid $smoke-white;
  border-radius: 6px;

  .base-container_title {
    @include container-header-text();
    margin-left: -20px;
    width: calc(100% + 40px);
    padding-left: 10px;

    & > .tooltip-icon {
      margin-top: 2px;
    }
  }
}
