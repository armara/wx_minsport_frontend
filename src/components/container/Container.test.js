import React from 'react'
import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import renderer from 'react-test-renderer'

import Container from 'components/container/Container'

configure({ adapter: new Adapter() })

describe('<Container />', () => {
  it('renders correctly default props', () => {
    const tree = renderer.create(<Container>test</Container>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
