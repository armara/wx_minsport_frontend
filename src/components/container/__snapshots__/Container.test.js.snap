// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly default props 1`] = `
<div
  className="base-container"
>
  
  test
</div>
`;
